src/main/java/app/controller/settings/Settings.java

```diff
@@ -35,12 +36,95 @@ public void addFurniture(Rectangle2D rectangle, FurnitureType type)
     {
-        furniture.add(new SettingsObject(rectangle, vector));
+        furniture.add(new SettingsObject(rectangle, vector, teleportRotation));
+    }
+
+    public void addTexture(Rectangle2D rectangle, Integer textureType, Integer textureOrientation)
+    {
+        furniture.add(new SettingsObject(rectangle, textureType, textureOrientation));
     }
 
     public void addSoundSource(Vector position, double amplitude){
         soundSources.add(new SoundSourceBase(position, amplitude));
     }
+
+    public SettingsObject getTargetArea() {
+        for(SettingsObject obj: furniture) {
+            if(obj.getType()==FurnitureType.TARGET)
+                return obj;
+        }
+        return null;
+    }
+
+    public SettingsObject getSpawnAreaIntruders() {
+        for(SettingsObject obj: furniture) {
+            if(obj.getType()==FurnitureType.INTRUDER_SPAWN)
+                return obj;
+        }
+        return null;
+    }
+
+    public SettingsObject getSpawnAreaGuards() {
+        for(SettingsObject obj: furniture) {
+            if(obj.getType()==FurnitureType.GUARD_SPAWN)
+                return obj;
+        }
+        return null;
+    }
+
+    public ArrayList<SettingsObject> getWalls() {
+        ArrayList<SettingsObject> objects = new ArrayList<>();
+        for(SettingsObject obj: furniture) {
+            if(obj.getType()==FurnitureType.WALL)
+                objects.add(obj);
+        }
+        return objects;
+    }
+
+    public ArrayList<SettingsObject> getTowers() {
+        ArrayList<SettingsObject> objects = new ArrayList<>();
+        for(SettingsObject obj: furniture) {
+            if(obj.getType()==FurnitureType.TOWER)
+                objects.add(obj);
+        }
+        return objects;
+    }
+
+    public ArrayList<SettingsObject> getPortals() {
+        ArrayList<SettingsObject> objects = new ArrayList<>();
+        for(SettingsObject obj: furniture) {
+            if(obj.getType()==FurnitureType.PORTAL)
+                objects.add(obj);
+        }
+        return objects;
+    }
+
+    public ArrayList<SettingsObject> getShades() {
+        ArrayList<SettingsObject> objects = new ArrayList<>();
+        for(SettingsObject obj: furniture) {
+            if(obj.getType()==FurnitureType.SHADE)
+                objects.add(obj);
+        }
+        return objects;
+    }
+
+    public ArrayList<SettingsObject> getTextures() {
+        ArrayList<SettingsObject> objects = new ArrayList<>();
+        for(SettingsObject obj: furniture) {
+            if(obj.getType()==FurnitureType.TEXTURE)
+                objects.add(obj);
+        }
+        return objects;
+    }
+
+    public ArrayList<SettingsObject> getGlass() {
+        ArrayList<SettingsObject> objects = new ArrayList<>();
+        for(SettingsObject obj: furniture) {
+            if(obj.getType()==FurnitureType.GLASS)
+                objects.add(obj);
+        }
+        return objects;
```

Comment:
This branch has inherited all of this code from #129 - still requires changes before it can be closed
Replace with something like:

```java
 public List<SettingsObject> getFurniture(FurnitureType type) 
 {
      furniture.stream()
               .filter(obj -> obj.getType() == type)
               .toList();  
 }
```

This replaces all of the get methods in one go